plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroHiveFileWriter.java

```diff
@@ -71,19 +74,24 @@ public AvroHiveFileWriter(
         verify(requireNonNull(fileSchema, "fileSchema is null").getType() == Schema.Type.RECORD, "file schema must be record schema");
         verify(inputColumnNames.size() == inputColumnTypes.size(), "column names must be equal to column types");
         // file writer will reorder input columns to schema, we just need to impute nulls for schema fields without input columns
-        ImmutableList.Builder<String> outputColumnNames = ImmutableList.<String>builder().addAll(inputColumnNames);
+        ImmutableList.Builder<String> outputColumnNames = ImmutableList.<String>builder();
         ImmutableList.Builder<Type> outputColumnTypes = ImmutableList.<Type>builder().addAll(inputColumnTypes);
-        Map<String, Schema> fields = fileSchema.getFields().stream().collect(Collectors.toMap(Schema.Field::name, Schema.Field::schema));
+        Map<String, String> lowerToGivenName = getLowerCaseToGivenFieldName(fileSchema);
+        Map<String, Field> fields = fileSchema.getFields().stream().collect(Collectors.toMap(Field::name, Function.identity()));
         for (String inputColumnName : inputColumnNames) {
-            Schema fieldSchema = fields.remove(inputColumnName);
-            if (fieldSchema == null) {
-                throw new AvroTypeException("File schema doesn't have input field " + inputColumnName);
+            Field field = fields.remove(inputColumnName);
+            if (field == null) {
+                if (!lowerToGivenName.containsKey(inputColumnName)) {
+                    throw new AvroTypeException("File schema doesn't have input field " + inputColumnName);
```

Comment:
If this could happen because the user has set up something incorrectly this exception is fine, if this would only happen because an invariant you're assuming about the code is not true you can make it a `verify()`

Reply:
This can happen where the user completely mismatches the Trino Table and Avro URL setup and doesn't have Hive populating the metastore.

Reply:
Like the naive setup for avro_schema_url and avro_schema_literal, has us put in a (dummy int) column a lot. Which totally fails when we go to use it (in new and old version of the writer) WITHOUT using some Hive writing to init the metastore